system-helper/flatpak-system-helper.c

```diff
@@ -224,6 +418,44 @@ handle_deploy (FlatpakSystemHelper   *object,
       return TRUE;
     }
 
+  for (guint i = 0; i < ongoing_pulls->len; i++)
+    {
+      FlatpakSystemHelperOngoingPull *pull_tmp;
+      g_autofree gchar *user_folder = NULL;
+
+      user_folder = g_file_get_path (g_file_get_parent (path));
+      pull_tmp = g_ptr_array_index (ongoing_pulls, i);
+      if (g_strcmp0(pull_tmp->user_folder_name, user_folder) == 0)
+        {
+          ongoing_pull = pull_tmp;
+
+	  g_signal_handler_disconnect (ongoing_pull->mount_monitor, ongoing_pull->mount_monitor_id);
+          ongoing_pull->mount_monitor_id = 0;
+
+	  if (umount (user_folder) == -1)
+            {
+              int errsv = errno;
+              g_set_error (&error, G_IO_ERROR, g_io_error_from_errno (errsv),
+                           "Unmounting :%s failed: %s\n", user_folder, g_strerror (errsv));
+              g_dbus_method_invocation_return_gerror (invocation, error);
+              return TRUE;
+            }
+
+	  if (ongoing_pull->bindfs)
+            {
+              g_subprocess_wait (ongoing_pull->bindfs, NULL, NULL);
+              g_subprocess_force_exit (ongoing_pull->bindfs);
```

Comment:
`wait()` blocks until the subprocess dies, at which point killing it is not necessary. We only need to kill it in the case where `umount` fails because it's in use, and we need to do that before waiting for it to terminate.